plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/session/CodeGenerationState.kt

```diff
@@ -86,6 +91,10 @@ class CodeGenerationState(
             logger.warn(e) { "$FEATURE_NAME: Code generation failed: ${e.message}" }
             result = Result.Failed
             failureReason = e.javaClass.simpleName
+            if (e is FeatureDevException) {
+                failureReason = e.reason()
```

Comment:
it is the same with L93. Why don't we use the error className as reason and message as description?

Reply:
> error className as reason

We do, but encoding it explicitly in `FeatureDevException.reason()` which contains the class name lookup. Here, 95 and 93 are redundant, but I figured being explicit for `FeatureDevException` would be most clear to a maintainer.

> message as description

`desc` (and `operation`) are fed into `reasonDesc` instead of `message`. I don't want to naively start logging message directly, as the cardinality may be too high. We can think of `desc` as a simplified or lower-cardinality message, explicitly for logging.